Setuptools support added

GrumpyOldTroll opened the pull request.

MaxF12 commented on README.md:
```diff
@@ -21,22 +21,39 @@ People interested in participating in TAPS can [join the mailing list](https://w
+- libpcre
 
-	./build_dependencies.sh
+~~~
```
```suggestion
Install requirements on Linux:
~~~
```

renghardt replied:
Agree with differentiating between Linux and macOS - but "apt-get" is actually specific to Debian-like Linuxes.
So I suggest to write "Linux (Debian)" here :)

MaxF12 commented on README.md:
```diff
@@ -21,22 +21,39 @@ People interested in participating in TAPS can [join the mailing list](https://w
+sudo apt-get update
+sudo apt-get install -y libpcre3-dev cmake
+sudo apt-get install -y autoconf automake libtool
+
```
```suggestion
~~~
Install requirements on MacOS:
~~~
brew install pcre cmake autoconf automake libtool
~~~
Setup virtual environment and install library:
~~~
```

renghardt commented on README.md:
```diff
@@ -21,22 +21,39 @@ People interested in participating in TAPS can [join the mailing list](https://w
+source ~/venv-taps/bin/activate
+
+INSTALL_PATH=${HOME}/local_install \
+  python setup.py build install
```
This should be consistent: Either "python3" or "python".
As our README examples use "python", I suggest we use "python" here as well.
We already state that we require Python3.7 or above, Python2 is soon going to be deprecated, so I think this works.

GrumpyOldTroll replied:
I wrote it this way because it's what I have to actually type. I think on the default installs for both the latest ubuntu and the latest macos, "python" still invokes python 2 unless you're inside a venv of some sort, so I thought a quick mention of how that happens might save someone a bit of time when they're willing to give it a try.  To me that sounds better than consistency, but I'm not opposed to removing the venv line and adding Python3.7+ to the list of requirements, so I did that.

renghardt commented on pytaps/transports.py:
```diff
@@ -121,7 +121,7 @@ class UdpTransport(TransportLayer):
     async def close(self):
         print_time("Closing connection.", color)
-        self.transport.close()
+        # self.transport.close()
```
Why is this commented out now?

GrumpyOldTroll replied:
I commented that because I hit an error and wasn't sure how to handle it, and then forgot.  The error looked like this, and went away after removing that line:

~~~
2019-11-02 17:15:02.077919: Received 125-byte datagram
received packet len 125
2019-11-02 17:15:03.103180: Received 125-byte datagram
received packet len 125
2019-11-02 17:15:04.095427: Received 125-byte datagram
2019-11-02 17:15:04.489903: leaving multicast session.
2019-11-02 17:15:04.490542: Closing connection.
Task exception was never retrieved
future: <Task finished coro=<UdpTransport.close() done, defined at /home/jholland/venv-taps/lib/python3.7/site-packages/pytaps-0.1-py3.7-linux-x86_64.egg/pytaps/transports.py:122> exception=AttributeError("'UdpTransport' object has no attribute 'transport'")>
Traceback (most recent call last):
  File "/home/jholland/venv-taps/lib/python3.7/site-packages/pytaps-0.1-py3.7-linux-x86_64.egg/pytaps/transports.py", line 124, in close
    self.transport.close()
AttributeError: 'UdpTransport' object has no attribute 'transport' 
~~~

I *think* this was just a mistake, and that the transport object intentionally doesn't have a transport member, and that the line should be removed instead of commented out?

MaxF12 replied:
The "transport" that gets closed there is different from UdpTransport object. I will rename it in the future to make it more clear. (Sadly asyncio has a class called transport that is doing something other than our TransportLayer classes. 
The problem here is that if the UDP connection gets initiated in the normal unicast way a aiotransport object gets created by the asyncio createDatagramEndpoint call (or more specifically in the connectionMade callback from that call). Since we neve have an aiotransport object when using multicast it is missing here when its supposed to be closed. The easiest solution would be to check if we have unicast and only close the aiotransport in that case. I assume all the cleanup that is required when closing the multicast connection is done in your multicast library anyway.

GrumpyOldTroll replied:
OK, I've re-added the transport.close call.  Interestingly, it no longer fails this morning.  I'm not convinced this is correct as it stands, but I'm convinced I probably shouldn't have touched it as part of this set of edits, and that it should be treated as a separate issue if I can reproduce it.  Thanks for clarifying.

GrumpyOldTroll commented:
Whoops, responded on email for some reason.  Added responses to the comments instead, and deleted this.

GrumpyOldTroll commented:
OK, I think I've addressed all the comments, in a perhaps not ideally organized sequence of commits, as I worked through my own confusion.  I'll mark them all resolved, and please take a new look at the README, and open a new comment if you see something still. (EDIT: and also the transports.py, where I just removed the commented code.)
